site/content/docs/browser-checks/partials-code-snippets.md

```diff
@@ -61,6 +64,111 @@ During execution, the code snippet will be inlined before the script is run.
+Node.js `module.exports` object, and you can use it to export helper functions or values. Here's an example:
+
+This is our snippet, we call it **log_me**.
+
+```js
+function logMe (data) {
+  console.log(data)
+}
+
+module.exports = {
+  logMe
+}
+```
+
+This is our browser script, using the snippet. We fly in a value for the `data` argument.
+
+```js
+const { logMe } = require('./snippets/log_me.js')
+logMe('hello world')
+```
+
+This will print `hello world` to the console. 
+
+## Legacy: Handlebars Syntax
+
+> We still support our legacy Handlebars syntax for referencing snippets in Browser check code but recommend using the **require**
+syntax discussed at the beginning of this document.
+
+
+You can reuse commonly used parts of code by referencing code snippets in any browser check. To do this, you use the
+[Handlebars](https://handlebarsjs.com/guide/partials.html) partial notation `{{> my_code_snippet }}`.
+
+This comes in very handy when you have multiple browser checks targeting the same site or web app that share a:
+
+- common login procedure
+- common navigation flow
+- common setup and/or teardown procedures
+
+Using partials is very straightforward. Any code snippets are just copied inline as is.
+
+### Example: GitHub login
+
+Say we want to validate some parts of the GitHub website only available to logged in users. We want to have separate, small
+browser checks to have granular feedback whether each part functions.
+
+1. Create a snippet named **github_login** in the "code snippets" section with the code below.
+
+   {{< tabs "github_login_legacy" >}}
+   {{< tab "Playwright" >}}
+```javascript
+const playwright = require('playwright')
+const browser = await playwright.chromium.launch()
+const page = await browser.newPage()
+
+await page.goto('https://github.com/login')
+await page.type('#login_field', process.env.GITHUB_USER)
+await page.type('#password', process.env.GITHUB_PWD)
+await page.click('[name="commit"]')
+```
+    {{< /tab >}}   
+    {{< tab "Puppeteer" >}}
+```javascript
+const puppeteer = require('puppeteer')     
+const browser = await puppeteer.launch()
+const page = await browser.newPage()
+    
+await page.goto('https://github.com/login')
+await page.type('#login_field', process.env.GITHUB_USER)
+await page.type('#password', process.env.GITHUB_PWD)
+await page.click('[name="commit"]')
+```
+    {{< /tab >}}
+    {{< /tabs >}}
+
+    Notice we are referencing the `GITHUB_USER` and `GITHUB_PWD` environment variables. All environment variables are available
+    in partials / snippets, just like in "normal" browser check scripts. Your snippet should look like the screenshot below.
+
+    ![browser check code snippet](/docs/images/browser-checks/code-snippet.png)
+
+2. Create a new browser check. Reference the code snippet you just created as a partial. After this, just continue with the normal check.
+   During execution, the code snippet will be inlined before the script is run.
```

Comment:
@tnolet is this the recommended usage we should lead with? vs recommending wrapping the snippet into a function block to be easily invoked at any point in the script, more similar to our previous Handlebars functionality? (Maybe I am misunderstanding what you meant in the example, in which case we have to make the flow clearer, specifically -where- one would add the require in the script. Ping me on Slack if that's too confusing.)

Reply:
What might be going on here is that I am misunderstanding your suggestion on how to use these because the example is too short. Too little context shown maybe.